scripts/_helpers.py

```diff
@@ -769,3 +773,29 @@ def filter_codes(c_list, iso_coding=True):
     full_codes_list = filter_codes(list(set(full_codes_list)), iso_coding=iso_coding)
 
     return full_codes_list
+
+
+# Function to get the last Git commit message
+def get_last_commit_message():
+    try:
+        # Run the Git command to get the last commit message
+        result = subprocess.run(
+            ["git", "log", "-1", "--pretty=format:%H %s"],
+            capture_output=True,
+            text=True,
+        )
+        return result.stdout.strip()
+    except Exception as e:
+        logger.warning(f"Error getting the last commit message: {e}")
+        return ""
+
+
+# Function to add the last commit to the config
+def update_config(config):
+    try:
+        # Insert the last commit message to config
+        config.update({"git_commit": get_last_commit_message()})
+    except Exception as e:
+        logger.warning(f"Error updating the config: {e}")
+
+    return config
```

Comment:
On second thought: is this function really needed?
I think we could simply add `config.update({"git_commit": get_last_commit_message()})` wherever we need update_config.

Is there a reason for having it as a function?
Note: the try except is not needed: `get_last_commit_message()` already has an inner try-except, so it should never fail.
Is there an additional reason to have this function?

Reply:
In PyPSA-Earth-Sec we need this structure to add the last commit message of the submodule pypsa-earth as well. 
Considering to leave it like this for having the same structure in pypsa-earth and pypsa-earth sec. What do you think?

Deleted the try except 👍